gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerViewModel.java

```diff
@@ -183,8 +185,18 @@
   }
 
   private static ImmutableSet<MapFeature> concatFeatureSets(
-      ImmutableSet<MapFeature> a, ImmutableSet<MapFeature> b) {
-    return a.<MapFeature>builder().addAll(b).build();
+      Object[] objects) {
+    ImmutableSet<MapFeature> combinedFeatureSet = ImmutableSet.<MapFeature>builder().build();
```

Comment:
Why call build here and on each iteration rather than just using the builder and calling `build()` at the end?

Reply:
beacuse combinedFeatureSet is updating itself with adding new items and along with previous ones. 
`combinedFeatureSet = ImmutableSet.<MapFeature>builder().addAll(a).addAll(combinedFeatureSet).build();`